js/filter.js

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-console */
+import {getFragment} from './fragment.js';
+import {shuffleArray, debounce} from './util.js';
+
+const galleryFiltersPanel = document.querySelector('.img-filters');
+const galleryFiltersForm = document.querySelector('.img-filters__form');
+const picturesContainer = document.querySelector('.pictures');
+const RANDOM_QTY = 10;
+const RERENDER_DELAY = 300;
+
+//Добавление выбранному фильтру класса активности
+function toggleButtonClass (evt, id) {
+  const filtersList = galleryFiltersForm.getElementsByClassName('img-filters__button');
+  Array.from(filtersList).forEach((element) => {
+    element.className = 'img-filters__button';
+    const tappedButton = galleryFiltersForm.querySelector(`#${id}`);
+    tappedButton.classList.add('img-filters__button--active');
+  });
+}
+
+//Сравнение соседних элементов массива, сортировка на убывание
+function compareComments (b, a) {
+  return a.comments.length - b.comments.length;
+}
+
+//Изменение отображения галереи миниатюр в зависимости от режима фильтрации
+function onFiltersButtonClick (evt, pictures) {
+  const filterId = evt.target.id;
+  toggleButtonClass(evt, filterId);
+  const oldPictures = picturesContainer.getElementsByClassName('picture');
+  Array.from(oldPictures).forEach((element) => element.remove());
```

Comment:
```suggestion
  oldPictures.forEach((element) => element.remove());
```
метод forEach есть на дом элементах